### PR TITLE
Mnodes

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/multi-nodes/multi-nodes-gpu.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                git fetch --all
+                git checkout {SHA}
+                pip install -e .
+                pip install torchelastic
+                export PL_RUNNING_SPECIAL_TESTS=1
+                DEFAULTS_FIRST="-m torchelastic.distributed.launch --nnodes=2 --nproc_per_node=1 --rdzv_id=s-{SHA}-{PYTHON_VERSION_DASH}-{PYTORCH_VERSION_DASH}-e --rdzv_backend=etcd --rdzv_endpoint={TCP_ADDRESS}  -m coverage run --source pytorch_lightning -m pytest --verbose --capture=no"
+                DEFAULTS_AFTER="-m torchelastic.distributed.launch --nnodes=2 --nproc_per_node=1 --rdzv_id=s-{SHA}-{PYTHON_VERSION_DASH}-{PYTORCH_VERSION_DASH}-e --rdzv_backend=etcd --rdzv_endpoint={TCP_ADDRESS}  -m coverage run --source pytorch_lightning -a -m pytest --verbose --capture=no"
+                # special tests list
+                python ${DEFAULTS_FIRST} ./tests/backends/test_multi_nodes_gpu.py::test_logging_sync_dist_true_ddp
+                python ${DEFAULTS_AFTER} ./tests/backends/test_multi_nodes_gpu.py::test__validation_step__log
```
Comment (Borda) [inline .github/multi-nodes/multi-nodes-gpu.yaml:40]: this shall be defined in extra file hold inside this repo

**Comment by Borda** [comment]
the yaml was ousourced

**Review by Borda — APPROVED** [review]

File: .github/workflows/ci_test-mnodes.yml [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        key: ${{ runner.os }}-pip-multi-node
+        restore-keys: |
+          ${{ runner.os }}-pip-
+
+    - name: Install dependencies
+      run: |
+        pip install awscli coverage
+        # todo
+        pip install git+https://${{ secrets.PL_GHOST_TOKEN }}@github.com/PyTorchLightning/lightning-dtrun.git@v0.0.3 -q --no-cache-dir
+        #pip install git+https://${{ secrets.PL_GHOST_TOKEN }}@github.com/PyTorchLightning/lightning-dtrun.git@mnodes -q --no-cache-dir
```
Comment (nmiculinic) [inline .github/workflows/ci_test-mnodes.yml:86]: remove commented line

File: .github/workflows/ci_test-mnodes.yml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+      run: |
+        curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_$(uname -s)_amd64.tar.gz" | tar xz -C /tmp
+        sudo mv /tmp/eksctl /usr/local/bin
+      shell: bash
+
+    - name: Create Gpu Node Pool
+      run: |
+        aws eks --region $REGION update-kubeconfig --name $AWS_CLUSTER
+        eksctl create nodegroup --name=${{ steps.job.outputs.ID }} --cluster=$AWS_CLUSTER --node-type=$NODE_TYPE  --nodes=$NODES
+        # eksctl create nodegroup --name=${{ steps.job.outputs.ID }} --cluster=$AWS_CLUSTER --managed --spot --node-type=$NODE_TYPE  --nodes=$NODES
```
Comment (nmiculinic) [inline .github/workflows/ci_test-mnodes.yml:119]: delete commented out line

File: .github/workflows/ci_test-mnodes.yml [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      if: always()
+      shell: bash
+      run: |
+        sleep 5
+
+    - name: Find ETCD TCP Address
+      id: tcp
+      shell: bash
+      run: |
+        echo "::set-output name=TCP_ADDRESS::$(kubectl logs etcd -n elastic-job | grep -Eo '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}:[0-9]{1,4}' | head -1)"
```
Comment (nmiculinic) [inline .github/workflows/ci_test-mnodes.yml:148]: cannot you just do `kubectl get svc <name> -o=jsonpath='{.spec.clusterIP}'` or something nicer than grepping stuff

File: .github/workflows/ci_test-mnodes.yml [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        sleep 400
+
+    - name: Listen to Jobs Logging
+      shell: bash
+      run: |
+        # todo: Enable automatic checking.
+        # while [ $i -lt $MAX_CHECKS ]; do ((i++)); if kubectl logs ${{ steps.job.outputs.ID_NAME }}-worker-0 -n elastic-job | grep -i "error\|failed"; then status_code=1 && break; elif kubectl logs ${{ steps.job.outputs.ID }}-worker-0 -n elastic-job | grep "TEST END"; then status_code=0 && break; else printf "." ; fi; sleep $CHECK_SPEEP; done && \
+        # echo "Done waiting. Job status code: $status_code" && \
+        kubectl logs ${{ steps.job.outputs.ID_NAME }}-worker-0 -n elastic-job > /tmp/full_output.txt
+        if grep -q 'END_TOKEN' /tmp/full_output.txt ; then csplit /tmp/full_output.txt '/END_TOKEN/'; else mv /tmp/full_output.txt xx00; fi && \
```
Comment (nmiculinic) [inline .github/workflows/ci_test-mnodes.yml:187]: ohh,, I remember this :)

**Review by nmiculinic — COMMENTED** [review]
Looks good to me, added a few minor comments

**Review by justusschock — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Pull request merged**